whatshap/phred_scores.py

```diff
@@ -0,0 +1,89 @@
+"""
+Takes as input the chromosome wise ref-read kmer pair counts from whatshap learn,
+calculates the probability for observing each ref-read kmer pair across the whole genome
+and outputs a table with corresponding phred-scores
+"""
+
+import argparse
+import math
+from pathlib import Path
+import csv
+
+
+def phred(inputfile, outputfile, epsilon_value, kmer_size):
+    probs = {}
+    probabilities = {}
+    seen_comb = {}
+    comb_count = {}
+    k = int(kmer_size)
+    total_comb = float(4**k)
+    epsilon = float(epsilon_value)
+    path = Path(inputfile)
+    glob_path = path.glob("*.txt")
+    writer = open(outputfile, "w")
+    # Iterate over each chromosome file
+    for file in glob_path:
+        counts_file = open(file, "r")
```

Comment:
Please use a context manager: `with open(file):` (to close the file automatically when no longer needed). See [the example in the csv.reader documentation](https://docs.python.org/3/library/csv.html#csv.reader).